File: typescript/examples/04-document-cite-check.ts
/**
 * Example 04 — Document cite check (upload → poll → report PDF)
 * Language: TypeScript (Node.js)
 *
 * Usage: npx tsx examples/04-document-cite-check.ts path/to/brief.pdf
 */
import { readFile } from "node:fs/promises";
import { basename } from "node:path";
import { writeFile } from "node:fs/promises";
import { LawDiverClient } from "../src/client.js";

const filePath = process.argv[2];
if (!filePath) {
  console.error("Usage: npx tsx examples/04-document-cite-check.ts <brief.pdf|docx>");
  process.exit(1);
}

const client = new LawDiverClient();
const bytes = await readFile(filePath);
const fileName = basename(filePath);
const mime = fileName.endsWith(".docx")
  ? "application/vnd.openxmlformats-officedocument.wordprocessingml.document"
  : fileName.endsWith(".doc")
    ? "application/msword"
    : "application/pdf";

console.log(`Uploading ${fileName} (${bytes.byteLength} bytes)…`);

const { job, report } = await client.citeCheckDocument(
  new Blob([new Uint8Array(bytes)], { type: mime }),
  fileName,
  { downloadReport: true },
);

console.log(`status: ${job.status}`);
console.log(`pages: ${job.pageCount} · citations: ${job.citationCount}`);
console.log(`counts:`, job.counts);
if (job.otherAuthoritiesFound) {
  console.log(`other authorities (not verified): ${job.otherAuthoritiesFound}`);
}

if (report) {
  const out = `cite-report-${job.jobId}.pdf`;
  await writeFile(out, Buffer.from(report));
  console.log(`wrote ${out}`);
}

File: python/examples/04_document_cite_check.py
"""Example 04 — Document cite check (Python).

Usage: python examples/04_document_cite_check.py path/to/brief.pdf
"""

from __future__ import annotations

import sys
from pathlib import Path

sys.path.insert(0, str(Path(__file__).resolve().parents[1]))

from lawdiver import LawDiverClient

if len(sys.argv) < 2:
    print("Usage: python examples/04_document_cite_check.py <brief.pdf|docx>")
    raise SystemExit(1)

path = Path(sys.argv[1])

with LawDiverClient() as client:
    print(f"Uploading {path.name} ({path.stat().st_size} bytes)…")
    result = client.cite_check_document(path, download_report=True)

job = result["job"]
print(f"status: {job['status']}")
print(f"pages: {job.get('pageCount')} · citations: {job.get('citationCount')}")
print(f"counts: {job.get('counts')}")
if job.get("otherAuthoritiesFound"):
    print(f"other authorities (not verified): {job['otherAuthoritiesFound']}")

report = result.get("report")
if report:
    out = Path(f"cite-report-{job['jobId']}.pdf")
    out.write_bytes(report)
    print(f"wrote {out}")
